From 259d92c62f450fc1dbe94dedda5aeb9f91f40e91 Mon Sep 17 00:00:00 2001
From: Gregory Schlomoff <gregory.schlomoff@gmail.com>
Date: Wed, 6 Apr 2011 09:41:39 +0700
Subject: [PATCH 5/5] KIMAP::LoginJob: Adding a method in LoginJob to get the last response from the server

---
 kimap/loginjob.cpp |    8 ++++++++
 kimap/loginjob.h   |   12 ++++++++++++
 2 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/kimap/loginjob.cpp b/kimap/loginjob.cpp
index 139f873..2b2d0ac 100644
--- a/kimap/loginjob.cpp
+++ b/kimap/loginjob.cpp
@@ -74,6 +74,7 @@ namespace KIMAP
 
       QString userName;
       QString password;
+      QString lastResponse;
 
       LoginJob::EncryptionMode encryptionMode;
       QString authMode;
@@ -253,6 +254,8 @@ void LoginJob::handleResponse( const Message &response )
     commandName = i18n("StartTls");
   }
 
+  d->lastResponse = response.toString();
+
   if ( d->authMode == QLatin1String( "PLAIN" ) && !response.content.isEmpty() && response.content.first().toString()=="+" ) {
     if ( response.content.size()>1 && response.content.at( 1 ).toString()=="OK" ) {
       return;
@@ -506,5 +509,10 @@ void LoginJob::connectionLost()
 
 }
 
+QString LoginJob::lastResponse() const
+{
+  Q_D(const LoginJob);
+  return d->lastResponse;
+}
 
 #include "loginjob.moc"
diff --git a/kimap/loginjob.h b/kimap/loginjob.h
index cd45194..279f503 100644
--- a/kimap/loginjob.h
+++ b/kimap/loginjob.h
@@ -70,6 +70,18 @@ class KIMAP_EXPORT LoginJob : public Job
     void setPassword( const QString &password );
 
     /**
+     * Returns the last response sent by the server.
+     *
+     * This is the greeting message in the case of a successful logging, or the error message if the
+     * login failed.
+     *
+     * It can be useful for logging purposes, but note that the content of this response is not
+     * defined by the IMAP protocol and is implementation-dependent. You should not assume anything
+     * about what the content of this response will be.
+     */
+    QString lastResponse() const;
+
+    /**
      * Set the encryption mode for the connection. In case an encryption mode is set, the caller
      * MUST check the encryptionMode() result after executing the job, to see if the connection is
      * encrypted or not (e.g handshaking failed).
-- 
1.7.4.msysgit.0

