From d7050a89602e8c2ae50053f495b36d6c10ee34c7 Mon Sep 17 00:00:00 2001
From: Gregory Schlomoff <gregory.schlomoff@gmail.com>
Date: Sat, 30 Jul 2011 16:50:28 -0700
Subject: [PATCH] Fixing compilation on Mac OS X

---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5cda72d..9c1753e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,7 @@ set(Boost_MINIMUM_VERSION "1.34.0")
 if (WIN32)
   find_package(Boost ${Boost_MINIMUM_VERSION}) # (jstaniek) temp. fix unless we have graph library, not just headers
 else (WIN32)
-  find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS graph)
+  find_package(Boost ${Boost_MINIMUM_VERSION})
 endif (WIN32)
 macro_log_feature(Boost_FOUND "Boost C++ Libraries" "Required by core components" "http://www.boost.org" TRUE ${Boost_MINIMUM_VERSION} "Boost must include the boost-graph library")
 
-- 
1.7.3.1.msysgit.0

